.devcontainer/Dockerfile

```diff
@@ -1,5 +1,7 @@
 # Use an official Rust image as a parent image
```

Comment:
why are we changing this file?

Reply:
this creates the dev container with the local version of rustposix rather than copying it. makes it useful without VScode too.